opal/corelib/runtime.js

```diff
@@ -124,21 +124,139 @@
   //     Opal.Object.$$scope.get("Array")
   //
   // If a constant with the given name cannot be found, then a dispatch to the
-  // class/module's `#const_method` is called, which by default will raise an
+  // class/module's `#const_missing` is called, which by default will raise an
   // error.
   //
   // @param name [String] the name of the constant to lookup
   // @return [Object]
   //
-  Opal.get = function(name) {
-    var constant = this[name];
+  Opal.get = function(const_name) {
+    return Opal.const_get([this], const_name, true, true);
```

Comment:
I think using `Opal` instead of `this` would be better. 

I don't have any proof but I sense it's easier to inline something for JS runtimes if it has no `this` in the function.